hvplot/converter.py

```diff
@@ -2592,10 +2592,13 @@ def labels(self, x=None, y=None, data=None):
         kdims, vdims = self._get_dimensions([x, y], [text])
         cur_opts, compat_opts = self._get_compat_opts('Labels')
         element = self._get_element('labels')
-        return (
-            element(data, kdims, vdims)
-            .redim(**self._redim)
-            .apply(self._set_backends_opts, cur_opts=cur_opts, compat_opts=compat_opts)
+        if self.by:
+            labels = Dataset(data).to(element, kdims, vdims, self.by)
+            labels = labels.layout() if self.subplots else labels.overlay(sort=False)
```

Comment:
Took this from
https://github.com/holoviz/hvplot/blob/85ccf13fba0442e8aaa5d765d766096516394ee9/hvplot/converter.py#L2401-L2405